lib/health_check.rb

```diff
@@ -0,0 +1,97 @@
+class HealthCheck
+  include ChefResource
+
+  OK = 'ok'.freeze
+  NOT_OK = 'not ok'.freeze
+  REACHABLE = 'reachable'.freeze
+  TIMEOUT = 'timeout'.freeze
+  UNREACHABLE = 'unreachable'.freeze
+  ERRORING = 'erroring'.freeze
+  AUTHERROR = 'authentication error'.freeze
+
+  attr_reader :status, :erchef, :postgres
+
+  def initialize
+    @status = OK
+    @erchef = { status: REACHABLE }
+    @postgres = { status: REACHABLE }
+  end
+
+  #
+  # Do a general health check.
+  #
+  def check
+    erchef_health
+    erchef_authentication
+    postgres_health
+    overall
+  end
+
+  #
+  # Shortcut to see if all is well
+  #
+  def ok?
+    @status == OK
+  end
+
+  private
+
+  #
+  # Try to ping erchef as an indicator of general health
+  #
+  def erchef_health
+    erchef_health_metric do
+      chef.get_rest '_status'
+    end
+  end
+
+  #
+  # Try to authenticate with erchef to prove that oc-id is actually functioning
+  #
+  # TODO: in order for this to work, we need a test user, with
+  # username/password stored in the vault and accessible in Settings
+  #
+  def erchef_authentication
```

Comment:
That makes sense. I didn't check everything the monitoring there was doing. I'm still debating if this should be something that should be placed in the oc-id status endpoint however. The endpoint has value without it, but I guess the question is how much work should a status endpoint be expected to do. If we have to go so far as to ensure a user is setup on erchef for the status endpoint to work, that feels a step too far, unless it's a default user that we know will be there on erchef. Maybe the answer here is not to add this to this endpoint but to make the messaging around oc-id more clear (or log better messages so it's easier to more quickly track this down in the future).